modules/oe_whitelabel_paragraphs/tests/src/Kernel/Paragraphs/CarouselTest.php

```diff
@@ -0,0 +1,205 @@
+    $this->container->get('router.builder')->rebuild();
+
+    $fixtures_path = \Drupal::service('extension.list.module')->getPath('oe_whitelabel_paragraphs') . '/tests/fixtures/';
+    // Create English files.
+    $en_file_1 = file_save_data(file_get_contents($fixtures_path . 'example_1.jpeg'), 'public://example_1_en.jpeg');
+    $en_file_1->setPermanent();
+    $en_file_1->save();
+    $en_file_2 = file_save_data(file_get_contents($fixtures_path . 'example_1.jpeg'), 'public://example_2_en.jpeg');
+    $en_file_2->setPermanent();
+    $en_file_2->save();
+
+    // Create Bulgarian files.
+    $bg_file_1 = file_save_data(file_get_contents($fixtures_path . 'example_1.jpeg'), 'public://example_1_bg.jpeg');
+    $bg_file_1->setPermanent();
+    $bg_file_1->save();
+    $bg_file_2 = file_save_data(file_get_contents($fixtures_path . 'example_1.jpeg'), 'public://example_2_bg.jpeg');
+    $bg_file_2->setPermanent();
+    $bg_file_2->save();
+
+    // Create a couple of media items with Bulgarian translation.
+    $media_storage = $this->container->get('entity_type.manager')
+      ->getStorage('media');
+    $first_media = $media_storage->create([
+      'bundle' => 'image',
+      'name' => 'First image en',
+      'oe_media_image' => [
+        'target_id' => $en_file_1->id(),
+      ],
+    ]);
+    $first_media->save();
+    $first_media_bg = $first_media->addTranslation('bg', [
+      'name' => 'First image bg',
+      'oe_media_image' => [
+        'target_id' => $bg_file_1->id(),
+      ],
+    ]);
+    $first_media_bg->save();
+    $second_media = $media_storage->create([
+      'bundle' => 'image',
+      'name' => 'Second image en',
+      'oe_media_image' => [
+        'target_id' => $en_file_2->id(),
+      ],
+    ]);
+    $second_media->save();
+    $second_media_bg = $second_media->addTranslation('bg', [
+      'name' => 'Second image bg',
+      'oe_media_image' => [
+        'target_id' => $bg_file_2->id(),
+      ],
+    ]);
+    $second_media_bg->save();
+
+    // Create a few Carousel items paragraphs with Bulgarian translation.
+    $items = [];
+    for ($i = 1; $i <= 4; $i++) {
+      $paragraph = Paragraph::create([
+        'type' => 'oe_carousel_item',
+        'field_oe_title' => 'Item ' . $i,
+        'field_oe_text' => $i % 2 === 0 ? 'Item description ' . $i : '',
+        'field_oe_link' => $i % 2 === 0 ? [
+          // Make sure that URI properly handled.
+          'uri' => $i === 4 ? 'route:<front>' : 'http://www.example.com/',
+          'title' => 'CTA ' . $i,
+        ] : [],
+        'field_oe_media' => $i % 2 !== 0 ? $first_media : $second_media,
+      ]);
+      $paragraph->save();
+      $paragraph->addTranslation('bg', [
+        'field_oe_title' => 'BG Item ' . $i,
+        'field_oe_text' => $i % 2 === 0 ? 'BG Item description ' . $i : '',
+        'field_oe_link' => $i % 2 === 0 ? [
+          'uri' => 'http://www.example.com/',
+          'title' => 'BG CTA ' . $i,
+        ] : [],
+      ])->save();
```

Comment:
I guess it's ok, but I'm never a fan of these complex creations, I would have just had explicit data added. It's ok anyway.

Reply:
y not my style either, but my tests where slow so I had to make up time with a lot of copy paste from oe_theme